### PR TITLE
(fix) Modified code to intercept the login and logout to open/close websockets

### PR DESCRIPTION
**Pull request by mdhaman**
[SD-3548](https://dev.sourcefabric.org/browse/SD-3548)

Close the websocket on session timeout or logout and open the websocket on login. This will stop the events getting fired on the client after session timeout or logout.

### CODE REVIEW

File: client/app/scripts/superdesk/auth/session-service.js [hunk 1]
```diff
@@ -63,9 +63,9 @@ define(['lodash'], function(_) {
             setToken(session.token);
             setSessionId(session._id);
             setSessionHref(session._links && session._links.self.href);
-
             this.identity = null;
             resolveIdentity(this.updateIdentity(identity));
+            $rootScope.$broadcast('login');
```
Comment (amagdas) [inline client/app/scripts/superdesk/auth/session-service.js]: Maybe create some constants for `login` and `logout`.

**Comment by amagdas** [comment]
LGTM except the minor comment :+1:

File: client/app/scripts/superdesk/auth/session-service.js (hunk 1 above)
Comment (mdhaman) [inline client/app/scripts/superdesk/auth/session-service.js]: cool.

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

File: client/app/scripts/superdesk/notification/notification.js [hunk 2]
```diff
@@ -69,6 +78,14 @@
         };
 
         connect();
+
+        $rootScope.$on('logout', function(event) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk/notification/notification.js]: could be just `$rootScope.$on('logout', disconnect);`

**Comment by petrjasek** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**